src/lexer/tokenizer.rs

```diff
@@ -0,0 +1,415 @@
+            '%' => {
+                let token = if self.is_match('=') {
+                    TokenType::ModEquals
+                } else {
+                    TokenType::Modulus
+                };
+                self.add_token(token)
+            }
+            '|' => {
+                let token = if self.is_match('=') {
+                    TokenType::PipeEquals
+                } else if self.is_match('|') {
+                    TokenType::Or
+                } else {
+                    TokenType::Pipe
+                };
+                self.add_token(token)
+            }
+            '&' => {
+                let token = if self.is_match('=') {
+                    TokenType::AmpersandEquals
+                } else if self.is_match('&') {
+                    TokenType::And
+                } else {
+                    TokenType::Ampersand
+                };
+                self.add_token(token)
+            }
+            '~' => self.add_token(TokenType::Tilde),
+            '>' => {
+                let token = if self.is_match('=') {
+                    TokenType::GreaterEquals
+                } else if self.is_match('>') {
+                    TokenType::RightShift
+                } else {
+                    TokenType::Greater
+                };
+
+                self.add_token(token)
+            }
+            '<' => {
+                let token = if self.is_match('=') {
+                    TokenType::LesserEquals
+                } else if self.is_match('<') {
+                    TokenType::LeftShift
+                } else if self.is_match('-') {
+                    TokenType::LeftArrow
+                } else {
+                    TokenType::Lesser
+                };
+
+                self.add_token(token)
+            }
+            ',' => self.add_token(TokenType::Comma),
+            '!' => {
+                let token = if self.is_match('=') {
+                    TokenType::BangEquals
+                } else {
+                    TokenType::Bang
+                };
+                self.add_token(token)
+            }
+            '$' => self.add_token(TokenType::Dollar),
+            '=' => {
+                let token = if self.is_match('=') {
+                    TokenType::Equals
+                } else {
+                    TokenType::Assign
+                };
+                self.add_token(token)
+            }
+            '.' => self.add_token(TokenType::Dot),
+            ':' => self.add_token(TokenType::Colon),
+            ';' => self.add_token(TokenType::Semicolon),
+            '?' => self.add_token(TokenType::QuestionMark),
+            '@' => self.add_token(TokenType::At),
+            '#' => self.add_token(TokenType::Hash),
+            '\'' => self.mutate(),
+            '"' => self.string(),
+            '`' => self.add_token(TokenType::BackTick),
+            'r' => {
+                self.add_token(TokenType::Register);
+                if !Tokenizer::is_digit(self.peek()) {
```

Comment:
If not a number after r, we should probably treat it as a original identifier.
Otherwise all identifiers starting with `r` would be disallowed.

Reply:
You are 100% right!